### ISSUE TITLE
Inserting array values

### ISSUE THREAD

**Issue by mmkal** [gajus/slonik#54]
Hi - I may be missing something but I don't see any way to build a query like this in the docs:
```sql
insert into post(tags) values('{"foo", "bar", "a \"quote-containing\" string"}');
```
where `tags` is an [array](https://www.postgresql.org/docs/9.1/arrays.html) field like `text[]` (e.g. created with `create table post(id serial primary key, tags text[])`)

I'd like to be able to insert array values from variables with something like:
```js
const myValues = [myValue1, myValues2, myValue3]
slonik.query(sql`insert into post(tags) values(${sql.arrayValues(myValues)})`)
```
which would generate the first query.

Right now in the absence of `sql.arrayValues` I'm having to use
```js
sql.raw(myValues.map(v => JSON.stringify(v)).join(', '))
```
which doesn't feel right at all.

`sql.valueList` produces `error: bind message supplies 3 parameters, but prepared statement "" requires 0`
`sql.array` is a completely different syntax that (I think) doesn't apply to `insert`.
`identifierList` actually works for most strings, but this is probably luck, since these are values, not identifiers, and it breaks for strings with quotes in them.

**Comment by gajus** [comment]
Have you seen [`sql.array`](https://github.com/gajus/slonik#slonik-query-building-sql-array)?

**Comment by mmkal** [comment]
Ah - yes, that was failing because I was forgetting the second argument (`memberType`).

Still might be nice to have a helper for creating the `'{"foo", "bar"}'` and/or `array['foo', 'bar']` syntax. Maybe it `sql.array` could do one of those when the member type isn't specified?

**Comment by gajus** [comment]
> Still might be nice to have a helper for creating the '{"foo", "bar"}' and/or array['foo', 'bar'] syntax. Maybe it sql.array could do one of those when the member type isn't specified?

What would be the use case?

It goes back to this discussion that I was having with @vitaly-t:

https://github.com/gajus/slonik/issues/52#issuecomment-486293911

tl;dr; I don't support any use of code generation that produces a dynamic number of value bindings because it prevents query grouping. Unless of course there are some clear benefits. In this case, it would be just laziness (?).

**Comment by gajus** [comment]
Closing this, but please continue discussion if you have further comments.

**Comment by mmkal** [comment]
The use case is pretty minor - just the type inference that postgres can do for you. When inserting it feels unnecessary to specify the column type when it's easy to infer. So basically, yes laziness! But you are probably right that the costs outweigh the benefits.
